pkg/flare/archive_kubeapiserver.go

```diff
@@ -0,0 +1,39 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package flare
+
+import (
+	"encoding/json"
+	"io"
+
+	apiutil "github.com/DataDog/datadog-agent/pkg/api/util"
+	autoscalingWorkload "github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload"
+)
+
+// GetAutoscalerList fetches the autoscaler list from the given URL.
+func GetAutoscalerList(remoteURL string) ([]byte, error) {
```

Comment:
Could we move this as a flare provider in a component ? Maybe `comp/autoscaling` ?

The goal long term is to migrate all the content from `pkg/flare` in the comp where they belong so we can include the flare mechanism in any binary without pulling a tone of dependencies.